hal/stm32g0.c

```diff
@@ -284,11 +291,48 @@ void hal_init(void)
     clock_pll_on(0);
 }
 
-void hal_prepare_boot(void)
+static void RAMFUNCTION hal_secure_boot(void)
```

Comment:
Just a preference here, feel free to ignore. I'd avoid the `hal_` prefix on static functions within the specific hal, because that may wrongly suggest that the function is part of the `hal.h` API